TheJumpCloud/jumpcloud-ADMU
# Cut 4355 improve revert account rules

## Issues
* [Cut-4355](https://jumpcloud.atlassian.net/browse/Cut-4355) - Cut-4355-Revert-Account-Rules-Improve

## What does this solve?
* This release fixes the issue migrating a JCUser with a localUsername where the progress form GUI gets stuck
* When migrating with a localUsername, the progress form UI Migration Details `NewLocalUsername` will show the localUsername instead of the JumpCloudUsername
* Added account merge tracker and added failure logic to prevent account reversion that may result to data loss
* Remove unused .EXE binary files
* Added fix to enable progress form action buttons when JCAgent install fails 
* Removed redundant `$jcUsername` variable when running Test-`JumpCloudUsername` function
## Is there anything particularly tricky?
N/A
## How should this be tested?
GUI Stuck Fix
1. Migrate using a JCUser with localUsername
2. Validate that ONLY the migrated user with localUsername is created. Check `lusrmgr.msc`
![image](https://github.com/user-attachments/assets/78155db4-5876-4e1e-84b5-4feda626a80e)


Merge Tracker
1. Migrate a user
2. Intentionally fail during the copy registry process by going to Registry Editor and selecting the created the created migration key in HKEY_USERS
3. This should fail the migration, validate that this does not delete the created user profile in `lusrmgr.msc`
![image](https://github.com/user-attachments/assets/94ab8b3c-bcb0-42be-b77e-c11ce9fc7c04)

JCAgent Install Fail
1. Use an invalid 40 char connect key then migrate
2. This should fail the migration and enable the action buttons in the bottom right of the progress form GUI
![image](https://github.com/user-attachments/assets/4456e629-ed42-4012-835b-dcaf9375dda9)

MTP Migration:
1. Migrate using an MTP org and user
2. No MTP selection popups should arise during migration
3. Validate the CLI version works as intended as well

## Screenshots
